File: ChiliCookoff2/src/app/PersonDTO.ts
import { PartyDTO } from './PartyDTO';
import { Party } from './Party';

export class PersonDTO {
    public Id: string;
    public UserId: string;
    public FirstName: string;
    public LastName: string;
    public EmailAddress: string;
    public EnterChili: boolean;
    public ChiliName: string;
    public PartyCode: string;
    public IsAccountProfileUpdate: boolean;
    public JoinedParties: PartyDTO[];
    public HostedParties: Party[];

    constructor(
        Id: string,
        UserId: string,
        FirstName: string,
        LastName: string,
        EmailAddress: string,
        EnterChili: boolean,
        ChiliName: string,
        PartyCode: string,
        IsAccountProfileUpdate: boolean,
        JoinedParties: PartyDTO[],
        HostedParties: Party[]
    ) {
        this.Id = Id;
        this.UserId = UserId;
        this.FirstName = FirstName;
        this.LastName = LastName;
        this.EmailAddress = EmailAddress;
        this.EnterChili = EnterChili;
        this.ChiliName = ChiliName;
        this.PartyCode = PartyCode;
        this.IsAccountProfileUpdate = IsAccountProfileUpdate;
        this.JoinedParties = JoinedParties;
        this.HostedParties = HostedParties;
    }
}

File: ChiliCookoff2/Models/PersonDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace ChiliCookoff2.Models
{
    public class PersonDTO : IValidatableObject
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string EmailAddress { get; set; }
        public bool EnterChili { get; set; }
        public string ChiliName { get; set; }
        public string PartyCode { get; set; }
        public bool? IsAccountProfileUpdate { get; set; }

        public List<PartyDTO> JoinedParties { get; set; }

        public List<Party> HostedParties { get; set; }

        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            if (EnterChili && string.IsNullOrEmpty(ChiliName))
                yield return new ValidationResult("Chili Name is required.");

            if ((!IsAccountProfileUpdate.HasValue || !IsAccountProfileUpdate.Value) && string.IsNullOrEmpty(PartyCode))
                yield return new ValidationResult("Party Code is required.");
        }
    }
}